lib/active_support/cache/dalli_store.rb

```diff
@@ -141,7 +135,17 @@ def write(name, value, options=nil)
         instrument(:write, name, options) do |payload|
           with do |connection|
             options = options.merge(:connection => connection)
-            write_entry(name, value, options)
+            # Support :race_condition_ttl by appending a secondary expiration to the value.
+            if options[:race_condition_ttl] &&
+              (ttl = options[:expires_in] || @options[:expires_in])
+              options[:expires_in] = ttl + options[:race_condition_ttl] * 2
```

Comment:
@wjordan Why are we adding twice the race_condition_ttl value to the expires_in here?